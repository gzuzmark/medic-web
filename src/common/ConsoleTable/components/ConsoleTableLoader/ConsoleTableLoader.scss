.ConsoleTableLoader {
  align-items: center;
  display: flex;
  width: 360px;
  &--center {
    left: 50%;
    position: absolute;
    transform: translateX(-50%);
    z-index: 0;
  }
}