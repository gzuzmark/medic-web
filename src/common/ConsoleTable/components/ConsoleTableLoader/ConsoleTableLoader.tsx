import * as React from "react";
import { TextBold1 } from '../../../ConsoleText';
import Loader from "../../../Loader/Loader";
import './ConsoleTableLoader.scss'

interface IPropsConsoleTableLoader {
    loading: boolean;
    center: boolean;
    style?: React.CSSProperties;
}

const ConsoleTableLoader: React.FC<IPropsConsoleTableLoader> = (props) => {
    if (!props.loading) {
        return null;
    }
    return (
        <div className={`ConsoleTableLoader ${props.center && 'ConsoleTableLoader--center'}`} style={{...props.style}}>
            <Loader />
            <TextBold1 style={{fontWeight: 'bold', marginLeft: 30}}>
                {props.children}
            </TextBold1>
        </div>
    )
};


export default ConsoleTableLoader;
