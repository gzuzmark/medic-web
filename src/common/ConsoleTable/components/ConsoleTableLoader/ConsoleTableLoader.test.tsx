import { shallow } from 'enzyme';
import 'jest-localstorage-mock';
import * as React from 'react';
import ConsoleTableLoader from './ConsoleTableLoader';

describe('ConsoleTableLoader Test',() => {
    let props: any;
    let mountedConsoleTableLoader: any;
    const getComponent = () => {
        if (!mountedConsoleTableLoader) {
            mountedConsoleTableLoader = shallow(
                <ConsoleTableLoader {...props}>Caja de Mensajes</ConsoleTableLoader>
            );
        }
        return mountedConsoleTableLoader;
    };

    beforeEach(() => {
        props = {
            center: false,
            loading: true
        };
        mountedConsoleTableLoader = undefined;
    });

    it("render: render ConsoleTableLoader", () => {
        const component = getComponent();
        expect(component).toMatchSnapshot();
    });

    it("render: render ConsoleTableLoader should have class center", () => {
        props = {
            center: true,
            loading: true
        };
        const component = getComponent().find('.ConsoleTableLoader--center');
        expect(component.length).toBe(1);
    });

    it("render: render ConsoleTableLoader should be null", () => {
        props = {
            center: false,
            loading: false
        };
        const component = getComponent().find('.ConsoleTableLoader');
        expect(component.length).toBe(0);
    });

});