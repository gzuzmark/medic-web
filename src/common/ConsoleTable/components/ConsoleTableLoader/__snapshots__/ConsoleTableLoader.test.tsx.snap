// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConsoleTableLoader Test render: render ConsoleTableLoader 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ConsoleTableLoader
    center={false}
    loading={true}
  >
    Caja de Mensajes
  </ConsoleTableLoader>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Loader
          height={100}
          style={
            Object {
              "top": -37,
            }
          }
          top={50}
        />,
        <styled.span
          style={
            Object {
              "fontWeight": "bold",
              "marginLeft": 30,
            }
          }
        >
          Caja de Mensajes
        </styled.span>,
      ],
      "className": "ConsoleTableLoader false",
      "style": Object {},
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "height": 100,
          "style": Object {
            "top": -37,
          },
          "top": 50,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Caja de Mensajes",
          "style": Object {
            "fontWeight": "bold",
            "marginLeft": 30,
          },
        },
        "ref": null,
        "rendered": "Caja de Mensajes",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Loader
            height={100}
            style={
              Object {
                "top": -37,
              }
            }
            top={50}
          />,
          <styled.span
            style={
              Object {
                "fontWeight": "bold",
                "marginLeft": 30,
              }
            }
          >
            Caja de Mensajes
          </styled.span>,
        ],
        "className": "ConsoleTableLoader false",
        "style": Object {},
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "height": 100,
            "style": Object {
              "top": -37,
            },
            "top": 50,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Caja de Mensajes",
            "style": Object {
              "fontWeight": "bold",
              "marginLeft": 30,
            },
          },
          "ref": null,
          "rendered": "Caja de Mensajes",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
